tools/mksurfdata_esmf/src/mksurfdata.F90

```diff
@@ -0,0 +1,1273 @@
+  call write_namelist_input()
+
+  ! ======================================================================
+  ! Create fsurdat
+  ! ======================================================================
+
+  ! Read in model mesh to determine the number of local points
+  mesh_model = ESMF_MeshCreate(filename=trim(mksrf_fgrid_mesh), fileformat=ESMF_FILEFORMAT_ESMFMESH, rc=rc)
+  if (ChkErr(rc,__LINE__,u_FILE_u)) call shr_sys_abort()
+
+  ! Get the number of local destination points on my processor (lsize_o)
+  call ESMF_MeshGet(mesh_model, numOwnedElements=lsize_o, rc=rc)
+  if (ChkErr(rc,__LINE__,u_FILE_u)) call shr_sys_abort()
+
+  ! Initialize urban dimensions (needed to initialize the dimensions in fsurdat)
+  call mkurbanInit(mksrf_furban)
+
+  ! Initialize pft/cft dimensions (needed to initialize the dimensions in fsurdat)
+  call mkpftInit( )
+
+  ! If fsurdat is blank, then we do not write a surface dataset - but we may still
+  ! write a dynamic landuse file. This is useful if we are creating many datasets at
+  ! once, and don't want duplicate surface datasets.
+  !
+  ! TODO(wjs, 2016-01-26) Ideally, we would also avoid doing the processing of
+  ! variables that are just needed by the surface dataset (not by the dynamic landuse
+  ! file). However, this would require some analysis of the above code, to determine
+  ! which processing is needed (directly or indirectly) in order to create a dynamic
+  ! landuse file.
+
+  ! Open fsurdat and write out variables
+  if (fsurdat == ' ') then
+     if (root_task) then
+        write (ndiag,'(a)') ' fsurdat is blank: skipping writing surface dataset'
+     end if
+  else
+     if (root_task)then
+        write(ndiag,*)
+        write(ndiag,'(1x,80a1)') ('=',k=1,80)
+     end if
+
+     ! Open file
+     ! TODO: what about setting no fill values?
+     call mkpio_wopen(trim(fsurdat), clobber=.true., pioid=pioid)
+
+     ! Define dimensions
+     call mkfile_define_dims(pioid, nx=mksrf_fgrid_mesh_nx, ny=mksrf_fgrid_mesh_ny, dynlanduse=.false.)
+
+     ! Define global attributes
+     call mkfile_define_atts(pioid, dynlanduse = .false.)
+
+     ! Define variables
+     call mkfile_define_vars(pioid, dynlanduse = .false.)
+
+     ! End define model
+     rcode = pio_enddef(pioid)
+  end if
+
+  ! NOTE: do not deallocate pctlak, pctwet, pctglacier and pcturban
+
+  ! -----------------------------------
+  ! Write out natpft, cft, and time
+  ! -----------------------------------
+  if (fsurdat /= ' ') then
+     rcode = pio_inq_varid(pioid, 'natpft', pio_varid)
+     rcode = pio_put_var(pioid, pio_varid, (/(n,n=natpft_lb,natpft_ub)/))
+     if (num_cft > 0) then
+        rcode = pio_inq_varid(pioid, 'cft', pio_varid)
+        rcode = pio_put_var(pioid, pio_varid, (/(n,n=cft_lb,cft_ub)/))
+     end if
+     ! time is months for LAI, SAI, and pft heights
+     rcode = pio_inq_varid(pioid, 'time', pio_varid)
+     rcode = pio_put_var(pioid, pio_varid, (/(n,n=n_jan,n_dec)/))
```

Comment:
Added these two lines to put values of 1 to 12 in the time variable (following cft just above as a template) but all I get is 12 zeros in my fsurdat's time variable.

@mvertens do you possibly see what I'm doing wrong?

Reply:
@jedwards4b provided a solution in #1695 with the explanation that time defined with an unlimited dimension needs to be written one value (ie one month) at a time.

@jedwards4b I brought the mods in manually. Also, just checking: You added a -g in the makefile. That's to compile in debug mode, no?

Thanks again!

Reply:
Right the -g was added because the way the build currently works the individual files are compiled with -g
but the link step is not.   I would like to work with you to improve the cmake build so that the second (src directory)
build step is no longer required.

Reply:
> Right the -g was added because the way the build currently works the individual files are compiled with -g but the link step is not.

So keeping the -g then, as well.

> I would like to work with you to improve the cmake build so that the second (src directory) build step is no longer required.

Sounds good. Shall we meet as a first step? Or could we even accomplish that during a meeting?

Reply:
> Right the -g was added because the way the build currently works the individual files are compiled with -g but the link step is not.

So keeping the -g then, as well.

> I would like to work with you to improve the cmake build so that the second (src directory) build step is no longer required.

Sounds good. Shall we meet as a first step? Or could we even accomplish that during a meeting?